functions/java/modules/module1/src/main/java/com/amazonaws/powertools/workshop/Utils.java

```diff
@@ -0,0 +1,183 @@
+package com.amazonaws.powertools.workshop;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.auth.signer.Aws4Signer;
+import software.amazon.awssdk.auth.signer.AwsSignerExecutionAttribute;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.GetItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.GetItemResponse;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+public class Utils {
+    private static final Logger LOGGER = LogManager.getLogger(Utils.class);
+    private static final String TABLE_NAME_FILES = System.getenv("TABLE_NAME_FILES");
+    private static final String BUCKET_NAME_FILES = System.getenv("BUCKET_NAME_FILES");
+    private static final String APPSYNC_ENDPOINT = System.getenv("APPSYNC_ENDPOINT");
+    private static final Region AWS_REGION = Region.of(System.getenv("AWS_REGION"));
+    private static final AwsCredentialsProvider credentialsProvider = EnvironmentVariableCredentialsProvider.create();
+
+    private Utils() {}
+    private static final S3Client s3Client = S3Client.builder()
+            .region(AWS_REGION)
+            .credentialsProvider(credentialsProvider)
+            .build();
+
+    public static final DynamoDbClient ddbClient = DynamoDbClient.builder()
+            .region(AWS_REGION)
+            .credentialsProvider(credentialsProvider)
+            .build();
+
+    public static Map<String, String> getImageMetadata(String key) {
+        String fileId = extractFileId(key);
+        GetItemResponse response = ddbClient.getItem(GetItemRequest
+                .builder()
+                .tableName(TABLE_NAME_FILES)
```

Comment:
Feels a bit janky to pull configuration in a utils class from an environment variable

Reply:
It's not the most beautiful code I wrote, I kept the same "structure" as typescript... If you want to refactor feel free, I have no more time to do this.